turbo/execution/eth1/forkchoice.go

```diff
@@ -468,3 +460,34 @@ func (e *EthereumExecutionModule) updateForkChoice(ctx context.Context, original
 		ValidationError: validationError,
 	})
 }
+
+func (e *EthereumExecutionModule) runPostForkchoiceInBackground(initialCycle bool) {
+	if e.doingPostForkchoice.Load() {
```

Comment:
```
if !atomic.Load() {
   return
} 
atomic.Store(true)
```
it's actually anti-pattern with race condition. 
need use:
```
if !atomic.CompareAndSwap(false, true) {
    return 
}
```

Reply:
done